src/Tiny_httpd.ml

```diff
@@ -697,8 +721,10 @@ module Sem_ = struct
   let release m t =
     Mutex.lock t.mutex;
     t.n <- t.n + m;
+    let r = t.n in
     Condition.broadcast t.cond;
-    Mutex.unlock t.mutex
+    Mutex.unlock t.mutex;
```

Comment:
OK, I just pushed a commit. I now have a function Sem_.available_connection which is separate and only
used if debug is activated.

Reply:
I am opening an issue about server status infos....

Reply:
oh, btw. We do have atomic read/writes for int, at least before multicore lands :grin: . Then this will need to be an `int Atomic.t`. I'll do that later but for now, you can remove the mutex.